src/SprykerSdk/SyncApi/OpenApi/Updater/OpenApiUpdater.php

```diff
@@ -0,0 +1,142 @@
+    }
+
+    /**
+     * @param \Transfer\UpdateOpenApiRequestTransfer $updateOpenApiRequestTransfer
+     *
+     * @return \Transfer\OpenApiResponseTransfer
+     */
+    public function updateOpenApi(UpdateOpenApiRequestTransfer $updateOpenApiRequestTransfer): OpenApiResponseTransfer
+    {
+        try {
+            $sourceOpenApi = Reader::readFromJson($updateOpenApiRequestTransfer->getOpenApiDocOrFail());
+        } catch (Throwable $throwable) {
+            return (new OpenApiResponseTransfer())
+                ->addError($this->messageBuilder->buildMessage($throwable->getMessage()));
+        }
+
+        if (!$sourceOpenApi->validate()) {
+            return (new OpenApiResponseTransfer())->setErrors(
+                new ArrayObject($sourceOpenApi->getErrors()),
+            );
+        }
+
+        $syncApiTargetFilepath = $this->filepathBuilder->buildSyncApiFilepath(
+            $updateOpenApiRequestTransfer->getOpenApiFileOrFail(),
+            $updateOpenApiRequestTransfer->getProjectRootOrFail(),
+        );
+
+        try {
+            if (is_file($syncApiTargetFilepath)) {
+                $targetOpenApi = Reader::readFromYamlFile($syncApiTargetFilepath, OpenApi::class, false);
+            } else {
+                $targetOpenApi = Reader::readFromYamlFile(
+                    $this->syncApiConfig->getPackageRootPath() . '/' .
+                    $this->syncApiConfig->getDefaultRelativePathToOpenApiFile(),
+                    OpenApi::class,
+                    false,
+                );
+            }
+
+            $targetOpenApi = $this->merge($targetOpenApi, $sourceOpenApi);
+
+            $this->createFileIfNotExists($syncApiTargetFilepath);
+            Writer::writeToYamlFile($targetOpenApi, $syncApiTargetFilepath);
+        } catch (Throwable $throwable) {
+            return (new OpenApiResponseTransfer())
+                ->addError($this->messageBuilder->buildMessage($throwable->getMessage()));
+        }
+
+        return (new OpenApiResponseTransfer())->addMessage($this->messageBuilder->buildMessage('Success!'));
+    }
+
+    /**
+     * @param \cebe\openapi\spec\OpenApi $targetOpenApi
+     * @param \cebe\openapi\spec\OpenApi $sourceOpenApi
+     *
+     * @return \cebe\openapi\spec\OpenApi
+     */
+    protected function merge(
+        OpenApi $targetOpenApi,
+        OpenApi $sourceOpenApi
+    ): OpenApi {
+        foreach ($this->mergerCollection as $merger) {
+            $targetOpenApi = $merger->merge($targetOpenApi, $sourceOpenApi);
+        }
+
+        return $targetOpenApi;
+    }
+
+    /**
+     * @param string $fileName
+     *
+     * @return void
+     */
+    protected function createFileIfNotExists(string $fileName): void
+    {
+        if (!is_file($fileName)) {
+            if (!is_dir(dirname($fileName))) {
+                mkdir(dirname($fileName), 0755, true);
+            }
+
+            file_put_contents($fileName, '');
+        }
```

Comment:
Wouldn't be

```suggestion
    if (!is_dir(dirname($fileName))) {
        mkdir(dirname($fileName), 0755, true);
    }

    file_put_contents($fileName, '');
```

enough?